### PR TITLE
Aether Scaling

### PR DESCRIPTION
**Pull request by mbilal92**
This PR introduces the following changes to support Aether scaling:

1. Enhanced error handling with RMutex for globalChunkTblMutex.
2. Replaced the fixed value of punchLivenessTime with a configurable variable.

### CODE REVIEW

File: drsm/api.go [hunk 1]
```diff
@@ -49,14 +49,19 @@ type DrsmInterface interface {
...
 	d := &Drsm{sharedPoolName: sharedPoolName,
 		clientId: myid,
 		db:       db,
 		mode:     ResourceClient}
-
+	// if punchLivenessTime == 0 {
+	// 	d.punchLivenessTime = 5000
+	// } else {
+	// 	d.punchLivenessTime = punchLivenessTime
+	// }
```
Comment (gab-arrobo) [inline drsm/api.go:64]: Why not removing this commented code? or why adding this code as comment?

File: drsm/updates.go [hunk 2]
```diff
@@ -185,6 +185,8 @@ func iterateChangeStream(d *Drsm, routineCtx context.Context, stream *mongo.Chan
 
 func (d *Drsm) punchLiveness() {
 	// write to DB - signature every 5 second
+	// or make it parameterized
+	// ticker := time.NewTicker(time.Duration(d.punchLivenessTime) * time.Millisecond)
```
Comment (gab-arrobo) [inline drsm/updates.go:189]: Why not removing this commented code? or why adding this code as comment?

File: drsm/updates.go [hunk 3]
```diff
@@ -200,6 +202,7 @@ func (d *Drsm) punchLiveness() {
 		//logger.AppLog.Debugf(" update keepalive time")
 		filter := bson.M{"_id": d.clientId.PodName}
 
+		// timein := time.Now().Local().Add(time.Millisecond * time.Duration(d.punchLivenessTime+500))
```
Comment (gab-arrobo) [inline drsm/updates.go:205]: Why not removing this commented code? or why adding this code as comment?

File: drsm/drsm.go [hunk 4]
```diff
@@ -58,7 +58,8 @@ type Drsm struct {
 	ipModule            ipam.Ipamer
 	prefix              map[string]*ipam.Prefix
 	mongo               *MongoDBLibrary.MongoClient
-	globalChunkTblMutex sync.Mutex
+	globalChunkTblMutex sync.RWMutex
+	punchLivenessTime   int
```
Comment (gab-arrobo) [inline drsm/drsm.go:62]: Why adding this variable if it is not used?

**Review by gab-arrobo — CHANGES_REQUESTED** [review]

File: drsm/api.go (hunk 1 above)
Comment (thakurajayL) [reply drsm/api.go:64]: I would like to setup a review call and go over the changes. PunchLiveNess is required functionality and its critical aspect of DRSM.

**Comment by thakurajayL** [comment]
Hi @mbilal92 , 
Please do following so it is easy to review your code
1. Fix conflict 
2. Update/rebase code so you can take new changes related to zap logging

File: drsm/drsm.go (hunk 4 above)
Comment (gab-arrobo) [inline drsm/drsm.go:62]: Commenting out this variable (which is unused) will fix the lint issue
```suggestion
	// punchLivenessTime   int
```

**Pull request closed without merging**